UnitsNet/CustomCode/UnitParser.cs

```diff
@@ -37,7 +37,8 @@ internal static class UnitParser
         internal static TUnit ParseUnit<TUnit>([NotNull] string str,
             [CanBeNull] IFormatProvider formatProvider,
             [NotNull] ParseUnit<TUnit> parseUnit,
-            [NotNull] Func<TUnit, TUnit, TUnit> add)
+            [NotNull] Func<TUnit, TUnit, TUnit> add,
+            Type unitType)
```

Comment:
I think I follow... see if the next commits are what you are expecting. [Another idea](https://github.com/plubbr/UnitsNet/commit/d20713528ef0d514db28cb61b0eab1c326a91ced) is to use an interface for quantity, then the `TUnit` argument can be used to determine the type with no need for a redundant parameter.